pkg/server/router.go

```diff
@@ -48,6 +48,9 @@ func router(ctx context.Context, config *Config, cfg *cmds.Server) http.Handler
 	authed.Path(prefix + "/server-ca.crt").Handler(fileHandler(serverConfig.Runtime.ServerCA))
 	authed.Path(prefix + "/config").Handler(configHandler(serverConfig, cfg))
 	authed.Path(prefix + "/readyz").Handler(readyzHandler(serverConfig))
+	authed.Path(prefix + "/encrypt/status").Handler(encryptionStatusHandler(serverConfig))
+	authed.Path(prefix + "/encrypt/toggle").Handler(encryptionToggleHandler(serverConfig))
+	authed.Path(prefix + "/encrypt/stage").Handler(encryptionStageHandler(serverConfig))
```

Comment:
Looking at this as we iterate, I think you can do this all in a single handler using a common data structure. Just put everything under `/secrets-encryption`. GET returns a struct with status of both encryption, and the rotation operation (if one is in progress). That same struct can be PUT to the endpoint to enable/disable encryption, or advance rotation between stages.
```suggestion
	authed.Path(prefix + "/encrypt/status").Handler(encryptionStatusHandler(serverConfig))
	authed.Path(prefix + "/encrypt/toggle").Handler(encryptionToggleHandler(serverConfig))
	authed.Path(prefix + "/encrypt/stage").Handler(encryptionStageHandler(serverConfig))
```

Reply:
I don't see a benefit to reducing it down to a single handler, it just makes the edge cases more dicey, and the single handler is just a massive IF statement. I am already using a common data structure for Toggle and Stage. The GET right now just returns a string output. Do we expect rancher to want to display the status? I was under the impression it was just for CLI operation (it wasn't in the original design doc).